### ISSUE TITLE
Not a Bug - Signal Community Mod Flags Posts - Are we suppressing free speech here? 

### ISSUE THREAD

**Issue by desmonk** [signalapp/Signal-Android#12061]
https://community.signalusers.org/t/its-2022-still-no-way-to-delete-contact/42682/1

This is kind of a hostage situation here. Developers have now really ignored this for so long that this even makes no sense to repeat as they are not going to so this unless they go back to their roots and begin taking privacy seriously again.

Imagine this.

You are having some sensitive conversation and don’t want to reveal the number of the person you’re talking to! This situation can be as simple as exchanging some personal messages to sharing some important revelation!

According to Signal, only way to delete is to block a contact but then again same contact stays there under a block list. Does this look like a solution? Transferring from conract list to block list is not a deletion.

Countless articles and requests are made for this feature to be introduced and all the developers have done nothing but to show us a middle finger labelling this as a feature request and not paying attention to serious consequences one can be put through for such lapse as depicted in above example.

There are many ways too. Delete the app! Uninstall and reinstall the app. They are workaround not solution! Nobody should have to do all such in order to delete a contact! A delete must mean delete. And a block doesn’t mean delete! Blocked can have their own block list but a delete must retain what it implies. Complete deletion of data.

You have no right to hold users contact info after a user makes a delete request. You’re holding data against their will!

To all the developers: Wake up!

Edit:
This genuine expressive post was flagged and and made hidden. This is now a dictatorship? This is a clear violation of free speech.

**Comment by Leptopoda** [comment]
Regarding your edit:
The community moderators are volunteers that are nit affiliated with signal.
We like to keep the forum tidy and have all the discussions in one place. Therefore we closed this thread as it was a duplicate of another mentioned discussion and also didn't bring any new information.

GitHub is only used for bug reports so your complaint seems to be in the wrong place.

**Comment by desmonk** [comment]
Not just that. The mod marked it as hidden! That is what pissed me off. Only wrote here as community itself is not run by open minded people who can't even stand some criticism and understand the intention.

That's why I titled this as not being a bug.

Also, you can too see how old this issue is! We are adding features over features and not a basic functionality like this?

Hope the mods at community are mature enough to not mark someone's legitimate arguments as hidden. They are abusing their power and nobody from Signal team doing anything about it.

**Comment by u32i64** [comment]
I've merged [your post](https://community.signalusers.org/t/15892/124) into the existing feature request topic.

I think this issue can now be closed.

In the future, if you are unsure why a moderator performed some action or don't agree with it, feel free to write a direct message to the `@moderators` group on the forum, the specific moderator who performed the action, or another moderator (see [About](https://community.signalusers.org/about) for a list). For hidden posts specifically, there's also a dedicated topic where many flags are explained: [Post hidden by community flags](https://community.signalusers.org/t/32336).

**Comment by alex-signal** [comment]
> This is a clear violation of free speech.

I don't think you understand free speech protections. Things like the 1st amendment, which only applies in the US, only protect you from the US Government suppressing your free speech, not from private entities moderating your content. Other countries have varying levels of free speech laws (like up in Canada, hate speech isn't protected) but it is still only a protection from Govt, not from private entities.

> According to Signal, only way to delete is to block a contact but then again same contact stays there under a block list.

I agree it isn't ideal, and don't know specifics (because I'm not a decision maker, I'm an app developer). Maybe there's a path where recipient rows can be "tombstoned" (where we delete identifying data and essentially treat them as "Unknown". 

Of course, if you delete the contact completely, we no longer can know that they're blocked. And as soon as they send you another message, we'll recreate that contact in your database. I'm not saying this isn't a solvable problem, but just want to say that this isn't as simple as deleting a row from a database.

**Comment by desmonk** [comment]
I am not referring to legality of free speech but how the action taken by mods were out of line to mark that as hidden as this issue is quite old and has its roots when app was known as Text Secure! Mod should have taken more liberal approach rather and could have had some healthy conversation instead of cutting it off.

About this feature, this is quite an important one. Delete should mean delete and users don't have to worry about it being held somewhere. Privacy! That's the fundamental thing Signal based upon and here this right here is not that. 

Biggest let down is that most people won't even know that this is what's happening behind the curtain. A normal user will never know that this kind of deletion is not actually deleting anything! It's your duty to make the app as transparent as possible. This is more like quicksand type of act. 

Countless features are already added and yet this basic one and literally something so important is still in the waiting list!

This is why the post at forums was written with a clear intention that this being neglected. You know, you say are not a decision maker and I understand but hope you take one step about this long pending issue to the table. 

Imagine seeing hundred of features and bugs getting fixed but such a crucial one never getting addressed!

Closing this here. And I really really wish someone can make this issue heard!

Again, a delete must mean delete! No residue left after. Just how we shred files. Even having it labeled as "Unknown" or anything is still leaves a trace behind.

Thank you for listening.
